bitrisescript/src/bitrisescript/bitrise.py

```diff
@@ -258,11 +276,39 @@ async def run_build(artifacts_dir: str, workflow_id: str, **build_params: Any) -
 
     log.info(f"Created new job for '{workflow_id}'. Slug: {build_slug}")
 
-    try:
-        await wait_for_build_finish(build_slug)
-        log.info(f"Build '{build_slug}' is successful. Retrieving artifacts...")
-        await download_artifacts(build_slug, artifacts_dir)
-    finally:
-        log.info(f"Retrieving bitrise log for '{build_slug}'...")
-        await download_log(build_slug, artifacts_dir)
-        await dump_perfherder_data(artifacts_dir)
+    await finalize_running_build(artifacts_dir, build_slug)
+
+
+async def get_running_builds(workflow_id: str, **kwargs) -> Optional[str]:
+    """Attempt to resume a bitrise build corresponding to the specified worlfow and build params.
+
+    Args:
+        artifacts_dir (str): Directory to download artifacts to.
+        workflow_id (str): Identifier of workflow to run.
+        kwargs (Any): Additional parameter to forward to bitrise.
+
+    Returns:
+        The list of running builds
+    """
+    client = BitriseClient()
+    data = {
+        "workflow": workflow_id,  # Note: I don't know why this API uses workflow instead of workflow_id
+        "status": 0,  # not finished (0), successful (1), failed (2), aborted with failure (3), aborted with success (4)
```

Comment:
I think we should also re-use successful builds if possible. E.g
1. Taskcluster task gets killed
2. Bitrise build finishes successfully
3. Taskcluster task is re-run
4. Taskcluster task finds successful build, downloads its artifacts and returns

Reply:
Mmmmm

> Taskcluster task gets killed
How would we differentiate from when we actually want a re-build?

ie: We have a green run on bitrise, but still want another build to be triggered.

Reply:
Good point.. I'm not sure

Reply:
For now I think we should only resume if it's currently running on bitrise. That way we can cancel a stuck workflow on the bitrise side if we really need to, and a rerun would fire up a new workflow.